lib/routes/exposures/index.js

```diff
@@ -113,37 +227,108 @@ async function exposures(server, options, done) {
         throw new Gone()
       }
 
-      const filteredExposures = exposures.filter(
-        ({ rollingStartNumber, rollingPeriod }) => {
-          const startTime = rollingStartNumber * 1000 * 600
-          const duration = rollingPeriod * 1000 * 600
+      resolvedOnsetDate = onsetDate
+    } else {
+      const { appPackageName, hmackey, verificationPayload } = request.body
 
-          return (
-            onsetDate === null ||
-            isAfter(new Date(startTime + duration), new Date(onsetDate))
-          )
+      const { reportType, symptomOnsetInterval, tekmac } = server.jwt.verify(
+        verificationPayload,
+        {
+          audience: options.exposures.certificateAudience,
+          issuer: options.security.jwtIssuer
         }
       )
 
-      if (filteredExposures.length > 0) {
-        await server.pg.write.query(
-          exposureInsert({
-            exposures: filteredExposures,
-            regions: regions || [options.exposures.defaultRegion]
-          })
+      const exposuresString = resolvedExposures
+        .sort((a, b) => a.keyData.localeCompare(b.keyData))
+        .map(
+          ({
+            keyData,
+            rollingStartNumber,
+            rollingPeriod,
+            transmissionRiskLevel
+          }) =>
+            `${keyData}.${rollingStartNumber}.${rollingPeriod}.${transmissionRiskLevel}`
+        )
+        .join(',')
+
+      const secret = Buffer.from(hmackey, 'base64')
+      const hash = crypto
+        .createHmac('sha256', secret)
+        .update(exposuresString)
+        .digest('base64')
+
+      if (
```

Comment:
Should we be doing the full set of checks here like we do on register?

Reply:
These are deprecated on the reference server, but I agree that the checks should be performed.

What value should we use as the nonce? I think the HMAC value is probably the best bet, it's returned from the API request prior to this one so is readily available. However, the nonce seems to get transformed by the SafetyNet API so we need to do the same transformations on our side before doing the equality check. I can't find any information about these transformations, so this will be trial and error.